arangod/VocBase/vocbase.cpp

```diff
@@ -1372,6 +1458,55 @@ arangodb::Result TRI_vocbase_t::dropCollection(DataSourceId cid,
   }
 }
 
+arangodb::Result TRI_vocbase_t::validateCollectionParameters(
+    arangodb::velocypack::Slice parameters) {
+  bool valid = parameters.isObject();
+  if (valid) {
```

Comment:
Should this return "BAD_PARAMETER" here, in case the parameters is not an Object? It will result in an error later, but "ILLEGAL_NAME" is not the "correct" error IMO.

Reply:
Added collection name to the message.